### PR TITLE
Slack Posting on New Implant

### PR DESCRIPTION
**Pull request by x41x41x41**
I've copied the pushover code to allow the same functionality within Slack. Currently uses bot tokens which will allow for extensions in the future.

I've tested
- Options when set via config file
- Setting same options via posh
- With and without a userid set
- Confirming the configuration appears using the `serverinfo` command

In addition, I've used the token of both services to detect which service is actually configured. The kill date change to the master branch also seems to have been pulled into this request....

Sorry if there's any .git related fopar, this is my first pull request... If there's anything I should change/do over let me know.

### CODE REVIEW

**Comment by benpturner** [comment]
Hey thanks John! This is looking great, will have a test properly over the next few days and report back if there's any bugs or comments on the PR. 

Really appreciate you taking the time to do this change. Will be great to have Slack as an alternative to Pushover.

File: poshc2/server/Implant.py [hunk 1]
```diff
@@ -71,23 +71,46 @@ def display(self):
...
                 output = conn.getresponse()
                 if output.status != 200:
                     data = output.read()
                     print("\nPushover error: ")
                     print(data)
         except Exception as e:
             print("Pushover send error: %s" % e)
+        try:
+            Slack_BotToken = select_item("Slack_BotToken", "C2Server")
+            if EnableNotifications.lower().strip() == "yes" and Slack_BotToken not in ("", None):
```
Comment (riskydissonance) [inline poshc2/server/Implant.py]: this can just be `and Slack_BotToken` which will check if it's a null or empty string :)

**Review by riskydissonance — COMMENTED** [review]
+1 Looks great, thanks! Will test it and approve

File: poshc2/server/Implant.py (hunk 1 above)
Comment (x41x41x41) [reply poshc2/server/Implant.py]: I'll sort this (and the same for pushover) out tonight and update the PR

Comment (riskydissonance) [reply poshc2/server/Implant.py]: You don't have to it'll work fine it's more just an FYI !

File: poshc2/server/Implant.py [hunk 2]
```diff
@@ -88,7 +88,7 @@ def display(self):
             print("Pushover send error: %s" % e)
         try:
             Slack_BotToken = select_item("Slack_BotToken", "C2Server")
-            if EnableNotifications.lower().strip() == "yes" and Slack_BotToken not in ("", None):
+            if EnableNotifications.lower().strip() == "yes" and Slack_BotToken:
```
Comment (x41x41x41) [inline poshc2/server/Implant.py:91]: Couldn't help myself (have done the same to pushover on line 74)

**Review by benpturner — APPROVED** [review]
Looks great!

**Pull request merged**